frontend/src/utils.js

```diff
@@ -104,9 +104,29 @@ function copyToClipboard(str) {
   return result;
 }
 
+const LOCALIZE_NUMBERS = !!(
+  typeof Intl === 'object' &&
+  Intl &&
+  typeof Intl.NumberFormat === 'function'
+);
+
+const UNITS = ['B', 'kB', 'MB', 'GB'];
+function bytes(num) {
+  const exponent = Math.min(Math.floor(Math.log10(num) / 3), UNITS.length - 1);
+  const n = Number(num / Math.pow(1000, exponent));
+  const nStr = LOCALIZE_NUMBERS
+    ? n.toLocaleString(navigator.language, {
+        minimumFractionDigits: 1,
+        maximumFractionDigits: 1
+      })
```

Comment:
First of all, never use `navigator.language`, use `navigator.languages` - this is a fallback chain of your preferred languages.

Now, what do you mean by "my second or third choice l20n fallback"? If I understand correctly this is a website, so it will get `navigator.languages` as a fallback chain from the user.
You just have to pass it to `toLocaleString` and it'll pick the best locale matching the list.